src/utils.jl

```diff
@@ -0,0 +1,215 @@
+function forwarddiff(a::AbstractArray{T,N}; dims::Int) where {T,N}
+    require_one_based_indexing(a)
+    1 <= dims <= N || throw(ArgumentError("dimension $dims out of range (1:$N)"))
+
+    d = similar(a)
+    r = axes(a)
+    r0 = ntuple(i -> i == dims ? UnitRange(1, last(r[i]) - 1) : UnitRange(r[i]), N)
+    r1 = ntuple(i -> i == dims ? UnitRange(2, last(r[i])) : UnitRange(r[i]), N)
+
+    d0 = ntuple(i -> i == dims ? UnitRange(last(r[i]), last(r[i])) : UnitRange(r[i]), N)
+    d1 = ntuple(i -> i == dims ? UnitRange(1, 1) : UnitRange(r[i]), N)
+
+    d[r0...] = view(a, r1...) .- view(a, r0...)
+    d[d0...] = view(a, d1...) .- view(a, d0...)

```

Comment:
This should be broadcasting:

```julia
    d[r0...] .= view(a, r1...) .- view(a, r0...)
    d[d0...] .= view(a, d1...) .- view(a, d0...)
```

Otherwise it will generate intermediate array `tmp = view(a, r1...) .- view(a, r0...)` and then assigned to `d[r0...]`.

I've fixed this in https://github.com/JuliaImages/ImageBase.jl/pull/11.

Reply:
OK! I'll fix it in the next PR.